src/lib/data/collections.ts

```diff
@@ -5,55 +5,82 @@ import { HttpTypes } from "@medusajs/types"
 import { getCacheOptions } from "./cookies"
 
 export const retrieveCollection = async (id: string) => {
-  const next = {
-    ...(await getCacheOptions("collections")),
-  }
+  try {
```

Comment:
These don't actually need the try-catch for things to work right? I would maybe exclude them from this PR (or commit) so it's clear what is a must-have and what is a nice-to-have